peps/pep-9999.rst

```diff
@@ -0,0 +1,300 @@
+    URL while also supporting a full table seemed redundant and confusing.
+
+This PEP exists to formalize the informal conventions that have arisen, as well
+as explicitly document ``Home-page`` and ``Download-URL`` as deprecated in
+favor of equivalent ``Project-URL`` representations.
+
+Deprecating ``Home-page`` and ``Download-URL``
+==============================================
+
+This PEP proposes that ``Home-page`` and ``Download-URL`` be considered
+**deprecated**. This deprecation has implications for both package metadata
+producers (e.g. build backends and packaging tools) and package indices
+(e.g. PyPI).
+
+.. _metadata-producers:
+
+Metadata producers
+------------------
+
+This PEP stipulates the following for metadata producers:
+
+* When generating metadata 1.2 or layer, producers **SHOULD** emit only
+  ``Project-URL``, and **SHOULD NOT** emit ``Home-page`` or ``Download-URL``
+  fields.
+* When generating ``Project-URL`` equivalents for ``Home-page`` and
+  ``Download-URL``, producers **MUST** use the
+  `label conventions <conventions-for-labels>` described below.
+
+These stipulations do not change the optionality of URL fields in core metadata.
+In other words, producers **MAY** choose to omit ``Project-URL`` entirely
+at their discretion.
+
+This PEP does **not** propose the outright removal of support for ``Home-page``
+or ``Download-URL``. However, see :ref:`future-considerations` for
+thoughts on how a new (as of yet unspecified) major core metadata version
+could complete the deprecation cycle via removal of these deprecated fields.
+
+.. _package-indices:
+
+Package indices
+---------------
+
+This PEP stipulates the following for package indices:
+
+* When interpreting a distribution's metadata of version 1.2 or later
+  (e.g. for rendering on a web page), the index **SHOULD** prefer
+  ``Project-URL`` fields as a source of URLs over ``Home-page`` and
+  ``Download-URL``, even if the latter are explicitly provided.
+
+* If a distributions's metadata contains only the ``Home-page`` and
+  ``Download-URL`` fields, the index **MAY** choose to ignore those fields
+  and behave as though no URLs were provided in the metadata. In this case,
+  the index **SHOULD** present an appropriate warning or notification to
+  the uploading party.
+
+  * The mechanism for presenting this warning or notification is not
+    specified, since it will vary by index. By way of example, an index may
+    choose to present a warning in the HTTP response to an upload request, or
+    send an email or other notification to the maintainer(s) of the project.
+
+* If a distribution's metadata contains both sets of fields, the index **MAY**
+  choose to reject the distribution outright. However, this is
+  **NOT RECOMMENDED** until a future unspecified major metadata version
+  formally removes support for ``Home-page`` and ``Download-URL``.
+
+These stipulations do not change the optionality of URL processing by indices.
```

Comment:
Ideally, package repositories would refrain from reinterpreting the URLs in previously uploaded packages in ways that lose information:

```suggestion

* Any changes to the interpretation of metadata of version 1.2 or later that
  results in previously recognised URLs no longer being recognised
  **SHOULD NOT** be retroactively applied to previously uploaded packages.

These stipulations do not change the optionality of URL processing by indices.
```